### PR TITLE
Fixes FHIR-29336

### PR DESCRIPTION
**Pull request by barabo**

### CODE REVIEW

**Comment by barabo** [comment]
This was discussed on 2021-01-22 and the notes were captured here:
https://docs.google.com/document/d/1dsVVnSAZpYCplpNYtWp6OO1rQNQ9zeglA086pZfIw2M/edit#heading=h.eg5ekx5qzx5s

TL;DR - the approved proposal was as follows:

Proposal: change the ui.launchActivity and ui.done response to
* Change "success" to a "status: LaunchStatusCode", a "code" where the choices are
  * success
  * error
* add a "launchStatusDetail: [Coding]" which can be populated when status is `error`. We can define a preferred binding to a LaunchStatusDetails valueset.
* Note that extensions are still allowed in the response

**Comment by jmandel** [comment]
For the states code, this PR shows an example payload using a FHIR CodeableConcept but I think the resolution was to use a FHIR code (i.e. a bare string in JSON rather than an object with an array of Codings).

For the launch status detail, did we actually agree on what the preferred value set should include? And generally FHIR would use a CodeableConcept rather than a Coding (or array of codings) for these kinds of details

File: fsh/ig-data/input/pagecontent/index.md [hunk 1]
```diff
@@ -337,19 +337,34 @@ targetWindow.postMessage({
...
-boolean `success` parameter and an optional `details` string:
+`status` parameter and an optional `statusDetail` coding:
 
 ```js
 clientAppWindow.postMessage({
   "messageId": "<some new uid>",
   "responseToMessageId": "<uid from the client's request>",
   "payload": {
-    "success": true,
+    "status": {
```
Comment (barabo) [inline fsh/ig-data/input/pagecontent/index.md:363]: After changing this to a `code` - should this be simplified to: `"status": "success"`?

Comment (jmandel) [reply fsh/ig-data/input/pagecontent/index.md:363]: That's right!

**Comment by jmandel** [comment]
From our discussion: using a CodeableConcept for error detail follows a common pattern in FHIR, aligning with http://hl7.org/fhir/operationoutcome-definitions.html#OperationOutcome.issue.details (for example). In general, the advantage here is for concepts that might exist at different levels of granularity, or might be coded in different ways; the CodeableConcept allows for >1 translations, and also allows for a simple "display" text string in the case where no coded error details are avaialbe.

**Pull request merged**